### PR TITLE
upkeep/369: Standardize PHPCS and PHPCompatibility GH Actions

### LINKED ISSUE #369: Update PHPCS and PHPCompatibility GitHub Actions
Standardize the PHPCS and PHPCompatibility GitHub Actions across all extensions by adding/updating the following:

- [ ] Use the same GitHub action with the same PHPCS config (rules) for all extensions when opening a PR
- [ ] Check support for PHP 7.4+ (or alternatively higher versions as the minimum if a specific extension needs to e.g. because an SDK or dependency requires it)
- [ ] Check all PHPCS rules only **changed lines** in the PR (phpcs.xml.dist)
- [ ] Check only PHPCompatibility rule for **all files** in the PR (phpcs-compat.xml.dist)
- [ ] Validate PHPCS checks passing before merging a PR

More details here: https://10upwoo.wordpress.com/2023/08/17/updating-phpcs-and-phpcompatibility-github-actions-for-quark-extensions/

### PR DESCRIPTION
**Pull request by faisal-alvi**
### All Submissions:

* [ ] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changes proposed in this Pull Request:

* Used the same GitHub action with the same PHPCS config (rules) which is used for all extensions when opening a PR
* Checking all PHPCS rules only changed lines in the PR
* Excluded PHPCompatibility ruleset from the phpcs config
* Check only the PHPCompatibility rule for all files in the PR
* Add/update and use `phpcs.xml.dist` and `phpcs-compat.xml.dist` config files.
* Validate PHPCS checks passing before merging a PR
* The ESLint workflow file is also updated to match w/ the PHPCS workflow file format.
* Updated npm run script for phpcompat

Closes #369.

### How to test the changes in this Pull Request:

All the GitHub Actions should be passed.

### Other information:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changelog entry

> Dev - Update PHPCS and PHPCompatibility GitHub Actions.

### CODE REVIEW

**Review by dkotter — COMMENTED** [review]
Overall this looks good but I don't think we should be bumping our minimum PHP version here. I believe the goal is to still ensure our minimum PHP version matches the minimum PHP version of the minimum Woo version we support (which is still 7.3). The know the linked issue and the P2 post mention 7.4 but I'd suggest we stick with 7.3 for now

File: .github/workflows/qit.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
       - name: Set PHP version
         uses: shivammathur/setup-php@v2
         with:
-          php-version: 7.4
+          php-version: 7.3
```
Comment (dkotter) [inline .github/workflows/qit.yml]: This needs to stay at 7.4, as that's the minimum required for QIT
```suggestion
          php-version: 7.4
```

**Review by dkotter — APPROVED** [review]

File: .gitattributes [hunk 2]
```diff
@@ -14,7 +14,7 @@ phpunit.xml.dist export-ignore
 .tx export-ignore
 webpack.config.js export-ignore
 postcss.config.js export-ignore
-phpcs.xml export-ignore
+phpcs.xml.dist export-ignore
```
Comment (dkotter) [inline .gitattributes:17]: We probably want to exclude the PHPCS compat config as well I think

File: composer.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "require-dev": {
         "phpunit/phpunit": "^8.0",
         "woocommerce/woocommerce-sniffs": "^0.1.3",
+		"sirbrillig/phpcs-changed": "^2.11.1",
```
Comment (dkotter) [inline composer.json]: Looks like spacing is a bit off here

File: phpcs-compat.xml.dist [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<ruleset name="PHPCompatibilityWP">
+    <rule ref="PHPCompatibility"/>
+
+    <!-- Files to check -->
+	<arg name="extensions" value="php"/>
```
Comment (dkotter) [inline phpcs-compat.xml.dist]: Looks like spacing is a bit off here

**Review by dkotter — CHANGES_REQUESTED** [review]

**Review by dkotter — APPROVED** [review]

File: .github/workflows/phpcs.yml [hunk 5]
```diff
@@ -1,30 +1,50 @@
...
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
```
Comment (jeffpaul) [inline .github/workflows/phpcs.yml:36]: Shouldn't this be changed lines and not changed files?

Comment (dkotter) [reply .github/workflows/phpcs.yml:36]: It uses the `changed-files` action to get a list of changed files and then passes those files into the `phpcs-changed` package (line 50 in this file) which then pulls the changed lines out of those files

**Comment by faisal-alvi** [comment]
Just added a new commit to bump `woocommerce-sniffs` to `1.0.0` ([internal slack discussion link](https://10up.slack.com/archives/C01UCQGPBDJ/p1696260699336079?thread_ts=1696009651.377169&cid=C01UCQGPBDJ))

**Pull request merged**